### ISSUE TITLE
Spatial Transformer Layer in CuDNNV5

### ISSUE THREAD

**Issue by sxjscience** [apache/incubator-mxnet#2175]
Is there any plan to support the spatial transformer layer?

**Comment by piiswrong** [comment]
I don't think any one is on it. cudnn lstm is of higher priority but no one is on it yet either

**Comment by tornadomeet** [comment]
@sxjscience @piiswrong i'll work on it soon, both with and without  cuDNN v5,  when i finish it, i'll give a pr~

**Comment by sxjscience** [comment]
@tornadomeet Great! You can contact me if you need any help! We can work together for this OP.

**Comment by tornadomeet** [comment]
@sxjscience OK~

**Comment by morningpig917** [comment]
Something strange about spatial transformer and haven't figured out why yet...
I am using STN to transform a 14x14 attention mask, but getting different training losses even I have fixed all random seeds, i.e., 
mx.random.seed(1234)
np.random.seed(1234)
random.seed(1234)

And I have confirmed that without STN(since STN only transforms the attention mask, the network is still valid if I remove this part) the training loss should be identical given the same parameters.

Could you give me some advice? Thanks!

**Comment by tornadomeet** [comment]
please post your code and log, another, you'd better open a new issue for your prob. thanks.
